contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1429,12 +1429,24 @@ pre_transform_target_entry(ResTarget *res, ParseState *pstate,
 		if (alias_len > 0)
 		{
 			char	   *alias = palloc0(alias_len + 1);
-			bool		dq = *colname_start == '"';
-			bool		sqb = *colname_start == '[';
-			bool		sq = *colname_start == '\'';
+			bool		dq = false;
+			bool		sqb = false;
+			bool		sq = false;
 			int			a = 0;
 			const char *colname_end;
 			bool		closing_quote_reached = false;
+			const char *colname_limit = pstate->p_sourcetext +
+				strlen(pstate->p_sourcetext) - alias_len;
+
+			while (colname_start < colname_limit &&
+				(*colname_start == '(' || isspace((unsigned char) *colname_start)))
+			{
+				colname_start++;
+			}
+
+			dq = *colname_start == '"';
+			sqb = *colname_start == '[';
+			sq = *colname_start == '\'';
```

Comment:
We need more tests combining with the existing logic, e.g., `select ( "field1   "   ) from select_clause_parens1;`

Reply:
Added tests combining parentheses with brackets and double quotes with spaces. Please note, that the logic for spaces for parentheses and for brackets/double quotes is different. For parentheses all spaces are stripped. For brackets and double quotes spaces are preserved in the field name (this logic remains the same as before).